src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1037,3 +1037,89 @@ the probability mass function of $z_i$ in the posterior.   These may
+
+If both $\Theta$ and $Z$ were continuous, Stan could be used to sample $M$ draws from the joint posterior $p_{\Theta, Z}(\theta, z)$ and then estimate $\mu$ with
+$$   
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M {g(\theta_i, z_i)}.
+$$
+Given $Z$ is discrete, however, Stan cannot be used to sample from the joint posterior because Stan's NUTS algorithm cannot compute the derivatives it needs for the discrete parameters. Instead, as outlined in the previous sections describing specific models, the user can first marginalize out $Z$ from the joint posterior to give the marginalized posterior $p_\Theta(\theta)$. This marginalized posterior can then be implemented in Stan as usual, and Stan will give samples $\{\theta_i\}_{i = 1}^M$ from the marginalized posterior. 
+
+Using only the samples $\{\theta_i\}_{i = 1}^M$, how can we estimate $\mathbb{E}(g(\Theta, Z))$? We can use a conditional estimator. That is, for each function $g$ of interest we can compute
+$$    
+h(\Theta) = \mathbb{E}(g(\Theta, Z) \mid \Theta)
+$$
+and then estimate $\mathbb{E}(g(\Theta, Z))$ with
+$$    
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M h(\theta_i).
+$$
+This estimator is justified by the law of iterated expectation, the fact that
+$$    
+\mathbb{E}(h(\Theta)) = \mathbb{E}(\mathbb{E}(g(\Theta, Z) \mid \Theta)) = \mathbb{E}(g(\Theta, Z)) = \mu.
+$$
+Using this marginalized estimator provides a way to estimate the expectation of any function $g(\cdot)$ for all combinations of discrete or continuous parameters in the model. However, it presents a possible new challenge: evaluating the conditional expectation $\mathbb{E}(g(\Theta, Z) \mid \Theta)$.
+
+Evaluating the conditional expectation {-}
+
+Fortunately, the discrete nature of $Z$ makes evaluating $\mathbb{E}(g(\Theta, Z) \mid \Theta)$ easy. The function $h(\Theta)$ can be written as:
+$$    
+h(\Theta)
+= \mathbb{E}(g(\Theta, Z) \mid \Theta) 
+= \sum_{k} g(\Theta, k) \textrm{Pr}(Z = k \mid \Theta),
+$$
+where we sum over the possible values of the latent discrete parameters. An essential part of this formula is the probability of the discrete parameters conditional on the continuous parameters, $\textrm{Pr}(Z = k \mid \Theta)$. More detail on how this quantity can be calculated is included below. Note that if $Z$ takes infinitely many values then computing the infinite sums will involve, potentially computationally expensive, approximation. 
```

Comment:
We have also used square brackets for "Pr[Z = k \mid \Theta]" and anywhere else the arguments are random variables.